conda_pack/core.py

```diff
@@ -447,11 +450,60 @@ def pack(name=None, prefix=None, output=None, format='infer',
             else:
                 raise CondaPackException("Unknown filter of kind %r" % kind)
 
-    return env.pack(output=output, format=format, arcroot=arcroot,
-                    dest_prefix=dest_prefix,
-                    verbose=verbose, force=force,
-                    compress_level=compress_level,
-                    zip_symlinks=zip_symlinks, zip_64=zip_64)
+    subenvirons = []
+    if recursive:
+        env_dir = os.path.join(env.prefix, 'envs')
+        if os.path.exists(env_dir):
+            for env_name in os.listdir(env_dir):
+                try:
+                    subenvirons.append(CondaEnv.from_prefix(os.path.join(env_dir, env_name)))
+                except CondaPackException as exc:
+                    print('Could not pack environment: %s: %s' % (env_name, exc))
+
+    # Ensure the prefix is a relative path
+    arcroot = arcroot.strip(os.path.sep)
+
+    # The output path and archive format
+    output, format = env._output_and_format(output, format)
+
+    if os.path.exists(output) and not force:
+        raise CondaPackException("File %r already exists" % output)
+
+    if verbose:
+        print("Packing environment at %r to %r" % (env.prefix, output))
+
+    fd, temp_path = tempfile.mkstemp()
+
+    try:
```

Comment:
@dsludwig it strikes me that what you've done here is basically copy the code from `CondaEnv.pack` here, and making `CondaEnv.pack` itself superfluous. I think it could go a long way to allay his concerns if we can do a better job of making this an incremental change to the code. Can refactor this a bit to, say, return as much of this functionality as possible back to `CondaEnv.pack`?